sale_partner_company/models/res_partner.py

```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    sale_company_id = fields.Many2one("res.company", string="Selling Company")
```

Comment:
Improvement: that would be interesting to group partners by selling company (search view addition), and so adding an index here.

Reply:
Group by selling company it can be interesting indeed.

Regarding the index, from my understanding as we have a low cardinality on  `sale_company_id` we'll not gain in performance.